osu.Game.Rulesets.Osu/Mods/OsuModSpunOut.cs

```diff
@@ -2,21 +2,41 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
+using System.Collections.Generic;
 using osu.Framework.Graphics.Sprites;
 using osu.Game.Graphics;
 using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects.Drawables.Pieces;
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModSpunOut : Mod
+    public class OsuModSpunOut : Mod, IApplicableToDrawableHitObjects
     {
         public override string Name => "Spun Out";
         public override string Acronym => "SO";
         public override IconUsage? Icon => OsuIcon.ModSpunout;
-        public override ModType Type => ModType.DifficultyReduction;
+        public override ModType Type => ModType.Automation;
         public override string Description => @"Spinners will be automatically completed.";
         public override double ScoreMultiplier => 0.9;
         public override bool Ranked => true;
         public override Type[] IncompatibleMods => new[] { typeof(ModAutoplay), typeof(OsuModAutopilot) };
+
+        public void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables)
+        {
+            foreach (var hitObject in drawables)
+            {
+                if (hitObject is DrawableSpinner spinner)
+                {
+                    spinner.Disc.Enabled = false;
+                    spinner.Disc.OnUpdate += d =>
+                    {
+                        if (d is SpinnerDisc s && s.Valid)
```

Comment:
I'd still like clarification as to why this type check is necessary.

Reply:
Null check to suppress CI warning.